### PR TITLE
[Compute] BREAKING CHANGE: vmss create: rename orchestration mode values

### PR DESCRIPTION
**Pull request by qwordy**
**Description**

Resolve https://github.com/Azure/azure-cli/issues/16666

Update parameter for VMSS Orchestration mode. We have deprecated the parameters 'ScaleSetVM, VM' Allowed values should now be 'uniform, flexible' with default Uniform.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Review by yungezz — APPROVED** [review]
is it necessary to include breaking change in title?

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 1]
```diff
@@ -548,8 +548,8 @@ def load_arguments(self, _):
...
         c.argument('computer_name_prefix', help='Computer name prefix for all of the virtual machines in the scale set. Computer name prefixes must be 1 to 15 characters long')
-        c.argument('orchestration_mode', help='Choose how virtual machines are managed by the scale set. In VM mode, you manually create and add a virtual machine of any configuration to the scale set. In ScaleSetVM mode, you define a virtual machine model and Azure will generate identical instances based on that model.',
-                   arg_type=get_enum_type(['VM', 'ScaleSetVM']), is_preview=True)
+        c.argument('orchestration_mode', help='Choose how virtual machines are managed by the scale set. In Uniform mode, you define a virtual machine model and Azure will generate identical instances based on that model. In Flexible mode, you manually create and add a virtual machine of any configuration to the scale set.',
```
Comment (ajjha-az) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:551]: Choose how virtual machines are managed by the scale set. In Uniform mode, you define a virtual machine model and Azure will generate identical instances based on that model. In Flexible mode, you manually create and add a virtual machine of any configuration to the scale set **_or generate identical instances based on virtual machine model defined for the Scale Set_**

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 2]
```diff
@@ -2598,7 +2598,7 @@ def _get_public_ip_address_allocation(value, sku):
                 master_template.add_resource(build_msi_role_assignment(vmss_name, vmss_id, identity_role_id,
                                                                        role_assignment_guid, identity_scope, False))
 
-    elif orchestration_mode.lower() == vm_str.lower():
+    elif orchestration_mode.lower() == flexible_str.lower():
         if platform_fault_domain_count is None:
             raise CLIError("usage error: --platform-fault-domain-count is required in VM mode")
```
Comment (ajjha-az) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py:2603]: Update VM mode to Flexible mode

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py]: Thanks. Updated.

File: src/azure-cli/azure/cli/command_modules/vm/_params.py (hunk 1 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py]: Thanks. Updated.

**Comment by qwordy** [comment]
> 
> 
> is it necessary to include breaking change in title?

Yes.

**Pull request merged**